### PR TITLE
MortyMed: Part 1

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
experimental: TIME FOR A MEDICAL OVERHAUL, STRAIGHT FROM MRTY!
experimental: spoiler alert, it's a hybrid of vg and paramed
experimental: ------------------------
experimental: Spacemen can now feel pain.
experimental: Tramadol, Morphine, or Oxycodone can negate or reduce pain.
experimental: Too much pain equals shock!
experimental: ------------------------
experimental: Dragging people on the floor can hurt them, if they're bleeding!
experimental: Use roller beds, or lockers!
experimental: ------------------------
experimental: Added wheelchairs. Sprites from /vg/.
experimental: ------------------------
experimental: You are able to still walk around and use items, albeit slowly, in very early crit (<10 below 0)
experimental: ------------------------
experimental: Losing blood may result in temporary paralysis and loss of balance.
experimental: Losing wayy too much blood may result in cardiac arrest.
/:cl:

contains lots of copy paste from vg and para

### CODE REVIEW

**Comment by Wilchenx** [comment]
Nerf bruise packs fixing mortal wounds

**Comment by MoreRobustThanYou** [comment]
how about just remove them entirely

**Comment by Wilchenx** [comment]
Make it so you can bandage a wound instead and it will heal, sort of like baymed does it.

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -724,10 +728,13 @@
...
 		else
-			if(health <= HEALTH_THRESHOLD_CRIT)
-				stat = SOFT_CRIT
-			else
-				stat = CONSCIOUS
+			switch(health)
+				if(HEALTH_THRESHOLD_CRIT to HEALTH_THRESHOLD_PRECRIT)
+					stat = PRE_CRIT
+				if(HEALTH_THRESHOLD_PRECRIT to HEALTH_THRESHOLD_CRIT)
+					stat = SOFT_CRIT
```
Comment (XDTM) [inline code/modules/mob/living/carbon/carbon.dm]: from A to B = precrit, from B to A = softcrit?

File: code/modules/mob/living/carbon/carbon_defines.dm [hunk 2]
```diff
@@ -59,3 +59,7 @@
 	var/next_hallucination = 0
 	var/cpr_time = 1 //CPR cooldown.
 	var/damageoverlaytemp = 0
+
+	var/pain_shock_stage = 0
+	var/pain_level = 0 //Gets overwritten every tick! If you want to add pain "resistance" or constain pain, see update_pain_level()
+	var/pain_numb = 0 // When this is set, the mob isn't affected by shock or pain.
```
Comment (XDTM) [inline code/modules/mob/living/carbon/carbon_defines.dm]: please make this a trait

Comment (MoreRobustThanYou) [reply code/modules/mob/living/carbon/carbon_defines.dm]: that's not a bool, that's like stuttering/slurring/eye_blurry/etc,  it gets decreased every tick

File: code/modules/mob/living/carbon/shock.dm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	0.7	* getToxLoss() + 		\
+	1.5	* getFireLoss() + 		\
+	1.2	* getBruteLoss() + 		\
+	1.7	* getCloneLoss()
+
+	for(var/datum/reagent/R in reagents.reagent_list)
+		pain_level -= R.pain_resistance
+
+	if(slurring) //I'm not sure why this is here.
+		pain_level -= 20
```
Comment (XDTM) [inline code/modules/mob/living/carbon/shock.dm]: then remove it

File: code/modules/mob/living/carbon/shock.dm [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+	return min(pain_level, 0)
+
+
+/mob/living/carbon/proc/handle_shock() //Currently only used for humans
+	update_pain_level()
+
+/mob/living/carbon/proc/has_painkillers()
+	return (reagents.has_reagent("oxycodone") || reagents.has_reagent("tramadol"))
```
Comment (XDTM) [inline code/modules/mob/living/carbon/shock.dm]: snowflake

File: code/modules/mob/living/carbon/carbon_defines.dm (hunk 2 above)
Comment (XDTM) [reply code/modules/mob/living/carbon/carbon_defines.dm]: ideally you want a status effect for that then, but a trait that does this is generally easier and more appropriate when using chems (see pax for example)

**Comment by XDTM** [comment]
I really like the idea of implementing pain and can't wait to implement it in our other systems.

File: code/modules/mob/living/carbon/carbon_defines.dm (hunk 2 above)
Comment (MoreRobustThanYou) [reply code/modules/mob/living/carbon/carbon_defines.dm]: done

**Comment by 81Denton** [comment]
I feel like morphine should also reduce pain

**Comment by Pacmandevil** [comment]
I don't think you realize how much of a pain in the ass bones are on baycode. And I don't say this as a TG contributer, just some shitcoder from above the northern border.

Bone repair surgery would work fucking horrible on tg, especially if you can't do it by yourself, as unga bunga boye with toolbox can just fuck you over and make you need surgery after like 2 hits... and I've yet to see a doctor do surgery at all on tg.

Shock is also the worst fucking thing about baymed, and only serves to fuck people over even more, without actually adding much substance to the game.

I don't play TG all that much anymore, but I can assure you that this will end in a shitheap regardless.

**Comment by XDTM** [comment]
There's no bone stuff in this PR

**Comment by Pacmandevil** [comment]
experimental: ------------------------
experimental: Spacemen now actually have bones.
experimental: Bone repair surgery is in!

**Comment by MoreRobustThanYou** [comment]
i haven't pushed it yet, @XDTM

**Comment by XDTM** [comment]
oh, i didn't refresh the page and i couldn't see the updated description

**Comment by 81Denton** [comment]
/tg/ rounds on average have 2 doctors of which neither know how to even spell surgery. Most of the time they can't even use sleepers properly or forget to take hardsuits off before cryo.
Now imagine 10 spessmen with shattered legs waiting for Smoothbrain MD to fix them up. The absolute horror

I'd still like to see this testmerged though, maybe docs might do their job for once?

**Comment by MoreRobustThanYou** [comment]
Actually, you know what, no bones. Bonecode is hard.

**Comment by XDTM** [comment]
It's kind of a vicious cycle, if doctors don't need to know/do anything to do their jobs you won't see many of them, and if you don't doctors you think that there are no doctors for anything complicated.

That said, i'd rather have medical changes one at a time, instead of dumping the whole thing, testmerging it a couple rounds, until it's abandoned because barely anybody knows what to do and everyone complains.

**Comment by Noahnoah55** [comment]
The people that do know what to do don't play doctor because it's not fun. This adds longevity to the role, hopefully leading to more people playing it in the future.

**Comment by YogsMcree** [comment]
If this gets ported, it needs to be done hand in hand with Head admin retroactive rule-rewrites. The current rules are NOT conducive of a medical system such as this. As much as I would LOVE to see it in the game, currently you can be beating to -190 hp for disarming someone once and its valid under the rules. We need rules that enforce the use of security and nonlethal methods for conflicts rather than feeding someone a toolbox, dropping them into shock with 5 broken bones, otherwise this will quickly get out of hand and unmerged

**Comment by XDTM** [comment]
Seems that bonebreaking won't be a part of this, which seems to be the main gripe against an advanced medical system

**Comment by Armhulen** [comment]
^^^^^^^^^^^^^

**Comment by YakumoChen** [comment]
Shock is kind of awful to deal with and essentially acts as crit all on its own that you have no way of actually fixing

Baymed isn't really complete without the full surgery experience however, but it's also hell to go through as stuff like surgery takes absolute ages (which itself is a meme on /vg/), even with the upgraded surgery tools /vg/ has.

**Comment by YakumoChen** [comment]
Also you may as well just title it "Baymed" and not some name as if you're coding something actually original.

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 5]
```diff
@@ -252,3 +252,15 @@
...
+	name = "Oxycodone"
+	id = "oxycodone"
+	results = list("oxycodone" = 3)
+	required_reagents = list("plasma" = 1, "ethanol" = 1, "tramadol" = 1)
+
+/datum/chemical_reaction/tramadol
+	name = "Tramadol"
+	id = "tramadol"
+	results = list("tramadol" = 3)
+	required_reagents = list("inaprovaline" = 1, "ethanol" = 1, "oxygen" = 1)
```
Comment (Anonmare) [inline code/modules/reagents/chemistry/recipes/medicine.dm]: Inaprovaline doesn't have a recipe, might wanna add one or change the recipe here

**Comment by Shapsy** [comment]
I'm wary about anything that makes surgery a common procedure on account of there only being 2-3 surgery places on station (including the holodeck and robotics, if they get dragged into doing surgeries for people), between the upwards of 50-80 crew members on high pop. The additions of pain and deeper bloodloss seem neat though.

Also Tylenol is a brand name I'm pretty sure

**Comment by YogsMcree** [comment]
I would also like to point out that these changes will further reinforce the "I am going to clone you. Fuck fixing that noise." mentality

**Comment by Wilchenx** [comment]
be me
be robo
someone comes in with a broken hand
saw it off with a chainsaw and replace it with a mechanical one
all is good, he's safe from mortymed now

**Comment by YakumoChen** [comment]
Yeah Tylenol is copyrighted (though that didn't stop Freon from being in the game and still named as such until it was removed). Acetaminophen is the actual well-known compound name you're thinking of.

**Comment by Anonmare** [comment]
Salicylic Acid and oxandrolone should have minor analgesic qualities to them in my own opinion, it makes them that slightly more superior to use over bicardine and kelotane. 

Styptic and silver sulf patches could spike somebody's pain upon application to reflect their flavour text

**Comment by XDTM** [comment]
Hits should cause temporary pain, and maybe some weapons could have a pain multiplier. There could be a disease symptom causing pain, maybe a chronic pain trauma as well.

**Comment by Iamgoofball** [comment]
@KorPhaeron requesting a close because MRTY is permabanned from our servers, cant even supervise the testmerge on this, and is permabanned on every tg downstream

**Comment by Wilchenx** [comment]
oofball

**Comment by YakumoChen** [comment]
MRTY has been making PRs and contributing despite his ban for a long time now and has better code then you despite it, goof

**Comment by Wilchenx** [comment]
>has better code than goof

he has broken so much shit in his code we started calling him Morty on coderbus

**Comment by Iamgoofball** [comment]
@YakumoChen I understand you're upset over monks but no, MRTY's code is worse.

**Comment by Armhulen** [comment]
@Iamgoofball every single one? can you list them

File: code/modules/mob/living/blood.dm [hunk 6]
```diff
@@ -55,17 +55,39 @@
...
+						Knockdown(75)
+					else
+						Stun(75)
+			if(BLOOD_VOLUME_SURVIVE to BLOOD_VOLUME_ARREST)
+				if(prob(45)) //it doesn't instantly happen, you still have a chance to start chugging iron or hook yourself to an IV if possible
+					if(stat == CONSCIOUS)
+						visible_message("<span class='danger'>[src] clutches at [p_their()] chest as if [p_their()] heart is stopping!</span>", "<span class='userdanger'>It feels as if your heart has stopped!</span>")
+					stop_sound_channel(CHANNEL_HEARTBEAT)
+					playsound_local(src, 'sound/effects/singlebeat.ogg', 100, 0)
+					set_heartattack(TRUE)
```
Comment (Iamgoofball) [inline code/modules/mob/living/blood.dm]: players threw a fit in my goonmed port when low blood could result in death why the fuck do you think people will like it here

File: code/modules/mob/living/carbon/carbon.dm [hunk 7]
```diff
@@ -666,6 +666,8 @@
 	//Fire and Brute damage overlay (BSSR)
 	var/hurtdamage = getBruteLoss() + getFireLoss() + damageoverlaytemp
 	if(hurtdamage)
+		if(has_trait(TRAIT_NUMB))
+			hurtdamage = max((hurtdamage - 20) / 2, 0) //Make the damage appear smaller than it really is
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/carbon.dm]: people threw a fit over miner's salve hiding the damage amount

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 8]
```diff
@@ -268,6 +268,7 @@
 		if(resting)
 			resting = 0
 			update_canmove()
+		reagents.add_reagent("paracetamol", 1) //the power of hugs reduces pain!
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/carbon_defense.dm]: undocumented and easy to spam change

File: code/modules/mob/living/carbon/shock.dm [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+/mob/living/carbon/proc/update_pain_level()
+	if(has_trait(TRAIT_NUMB))
+		return 0 //No this isn't a bool
+
+	pain_level = 					\
+	1	* getOxyLoss() + 		\
+	0.7	* getToxLoss() + 		\
+	1.5	* getFireLoss() + 		\
+	1.2	* getBruteLoss() + 		\
+	1.7	* getCloneLoss()
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/shock.dm]: what the absolute fuck is this layout

File: code/modules/mob/living/carbon/shock.dm [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+	for(var/datum/reagent/R in reagents.reagent_list)
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/shock.dm]: typelesss lists

File: code/modules/mob/living/carbon/shock.dm (hunk 4 above)
Comment (Iamgoofball) [reply code/modules/mob/living/carbon/shock.dm]: snowflake and also typical baycode

File: code/modules/mob/living/living.dm [hunk 11]
```diff
@@ -261,7 +261,7 @@
 		return 1
 
 /mob/living/proc/InCritical()
-	return (health <= HEALTH_THRESHOLD_CRIT && (stat == SOFT_CRIT || stat == UNCONSCIOUS))
+	return (health <= HEALTH_THRESHOLD_CRIT && (stat == SOFT_CRIT || stat == UNCONSCIOUS || stat == PRE_CRIT))
```
Comment (Iamgoofball) [inline code/modules/mob/living/living.dm]: clean this shit up

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 12]
```diff
@@ -1274,4 +1274,64 @@
...
+		C.pain_shock_stage -= 3 //We don't FEEL the shock now, but make it go away quick in case we run out of oxycodone.
+		if(!C.IsSleeping() && !C.IsUnconscious() && prob(2))
+			to_chat(M, pick("<span class='numb'>You feel like you're floating...</span>", \
+							"<span class='numb'>You feel a little lightheaded... but it's okay.</span>", \
+							"<span class='numb'>Your face itches a little bit... and it feels so good to scratch it...</span>", \
+							"<span class='numb'>Your whole body buzzes slightly, but it doesn't seem to bother you...</span>", \
+							"<span class='numb'>You feel a little high of energy, and it makes you smile...</span>", \
+							"<span class='numb'>You nod to yourself... it's nothing, it just feels good to nod a little...</span>", \
+							"<span class='numb'>Hello?... Is there anybody in there?...</span>", \
+							"<span class='numb'>You feel... comfortably numb.</span>"))
```
Comment (Iamgoofball) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: make this a list

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 13]
```diff
@@ -1274,4 +1274,64 @@
...
+
+
+/datum/reagent/medicine/oxycodone
+	name = "Oxycodone"
+	id = "oxycodone"
+	description = "An effective and very addictive painkiller."
+	metabolization_rate = 0.05 * REAGENTS_METABOLISM
+	color = "#C805DC"
+	taste_description = "bitterness"
+	addiction_threshold = 15
```
Comment (Iamgoofball) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: no actual defined addiction steps

no downsides to this either, I can just ride high on oxy all fucking shift and nullify everything you did here

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 14]
```diff
@@ -252,3 +252,15 @@
 	results = list("modafinil" = 5)
 	required_reagents = list("diethylamine" = 1, "ammonia" = 1, "phenol" = 1, "acetone" = 1, "sacid" = 1)
 	required_catalysts = list("bromine" = 1) // as close to the real world synthesis as possible
+
+/datum/chemical_reaction/oxycodone
+	name = "Oxycodone"
+	id = "oxycodone"
+	results = list("oxycodone" = 3)
+	required_reagents = list("plasma" = 1, "ethanol" = 1, "tramadol" = 1)
```
Comment (Iamgoofball) [inline code/modules/reagents/chemistry/recipes/medicine.dm]: lmao shit this is easy to make

File: code/modules/vehicles/wheelchair.dm [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+		. += 1
+	if(C.get_bodypart("r_arm"))
+		. += 1
+
+/obj/vehicle/ridden/wheelchair/driver_move(mob/user, direction)
+	var/hands = check_hands(user) //See check_hands() proc above
+	if(hands <= 0)
+		return FALSE
+	movedelay = normal_move_delay * (4 / hands)
+	vehicle_move(direction)
```
Comment (Iamgoofball) [inline code/modules/vehicles/wheelchair.dm]: only good thing in this PR

**Review by Iamgoofball — CHANGES_REQUESTED** [review]
overall you not understanding how baymed works resulted in you fucking up this baymed port

File: code/modules/vehicles/wheelchair.dm (hunk 15 above)
Comment (Armhulen) [reply code/modules/vehicles/wheelchair.dm]: lol

**Comment by Iamgoofball** [comment]
@Armhulen he clarified in irc he's not banned from ftl apparently

**Comment by Armhulen** [comment]
when did he get banned from hippie

**Comment by Armhulen** [comment]
also wheelchairs don't work right grrrrrrr

**Comment by Armhulen** [comment]
THIS ISN'T CODE >:(

**Comment by praisenarsie** [comment]
@Armhulen when he fucked up the codebase and/or server like three times

**Comment by Armhulen** [comment]
explain

**Comment by MoreRobustThanYou** [comment]
salty admin and prick host happened

**Comment by Iamgoofball** [comment]
i think trying to add *poo didnt help either

**Comment by Armhulen** [comment]
doesn't explain much

File: code/modules/mob/living/blood.dm (hunk 6 above)
Comment (XDTM) [reply code/modules/mob/living/blood.dm]: Isn't this a bit above the instant death threshold anyway? You're already taking massive oxyloss by this point

File: code/modules/mob/living/carbon/carbon.dm (hunk 7 above)
Comment (XDTM) [reply code/modules/mob/living/carbon/carbon.dm]: what, when? i remember complaints about the instastun, but i don't think hiding damage was ever an issue

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 8 above)
Comment (XDTM) [reply code/modules/mob/living/carbon/carbon_defense.dm]: agree it shouldn't use a chem

**Comment by ExcessiveUseOfCobblestone** [comment]
@MoreRobustThanYou stop evading shitler

**Comment by praisenarsie** [comment]
@MoreRobustThanYou stop evading shitler

**Comment by Qustinnus** [comment]
cant be banned from ftl if he hasnt been on there, he wouldnt last for a weekend on ftl

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 13 above)
Comment (MoreRobustThanYou) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: it's supposed to be a warning like "hey dipshit you're getting close to overdose"

**Comment by Armhulen** [comment]
found him evading today no more testmerges

**Comment by praisenarsie** [comment]
@Armhulen are you saying this unironically or ironically or "unironically"

**Comment by Armhulen** [comment]
unironically

**Comment by Armhulen** [comment]
embarrassing innit

**Comment by MoreRobustThanYou** [comment]
seems more ""unironically"" due to radio silence on discord

**Comment by Armhulen** [comment]
:woman_facepalming:

**Comment by Plapatin** [comment]
>pain mechanics on /tg/ out of all places
why must you do this, everybody knows this is powergame central

**Comment by Armhulen** [comment]
lol rekt

**Comment by MoreRobustThanYou** [comment]
capture

**Comment by Armhulen** [comment]
he was banned for that very out of place

**Comment by Shadowflame909** [comment]
DARN YOU MRTY! >MRTY MED BROKE XENOBIO BLOOD REACTIONS. HOW TO TEST OUT. >testmerge morty med, get blood in a syringe. >inject it into any slime extract except for grey. >the slime extract will not do anything. For some reason grey works, but nothing else does with blood. DARN YOU, REEE

**Pull request closed without merging**